### DISCUSSION TITLE
Maddy does not support "UidPlus"?

### DISCUSSION THREAD

**Discussion by HubKing** [emersion/maddy#553] [Q&A] [answered]
I am using C# and the MailKit library to delete e-mails in the IMAP server.  The [sample code](https://stackoverflow.com/a/24204804/455796) is

```
if (client.Capabilities.HasFlag (ImapCapabilities.UidPlus))
    client.Inbox.Expunge (new UniqueId[] { uid });
```

but the capabilities I get is `MailKit.Net.Imap.ImapCapabilities.IMAP4rev1 | MailKit.Net.Imap.ImapCapabilities.Status | MailKit.Net.Imap.ImapCapabilities.LiteralPlus | MailKit.Net.Imap.ImapCapabilities.Idle | MailKit.Net.Imap.ImapCapabilities.Namespace | MailKit.Net.Imap.ImapCapabilities.Children | MailKit.Net.Imap.ImapCapabilities.Unselect | MailKit.Net.Imap.ImapCapabilities.SaslIR | MailKit.Net.Imap.ImapCapabilities.Sort | MailKit.Net.Imap.ImapCapabilities.Thread | MailKit.Net.Imap.ImapCapabilities.I18NLevel | MailKit.Net.Imap.ImapCapabilities.Move | MailKit.Net.Imap.ImapCapabilities.AppendLimit`, so, it seems that I cannot delete e-mails using the Expunge method. 

Does Maddy really not support UidPlus, or have I misconfigured it? The documentation does not have anything when I searched it for "uid".

**Comment by foxcpp** [comment, marked as answer]
Yes, you are right. maddy does not support the UIDPLUS extension.

**Reply by HubKing** [reply]
I don't know much about IMAP, but as the author of MailKit says, I kind of feel it is unsafe to delete e-mails using their indices rather than unique ID's, because indices can change. I don't know why it was not designed to use unique ID's in the first place and only supports it through an "extension".
